SE-XXX Typed throws.md

```diff
@@ -502,6 +500,83 @@ In first place we will specify different examples where the Swift compiler inter
 
 `// TODO: I think we should put that to the detailed solution part now`
 
+### Reducing `catch` clause verbosity
+
+`// TODO: This is the \catch\ proposal, to be determined whether it should be includded in the final draft or not`
+
+From the side of the `catch` clause there are becoming verbose due to the fact of being the place where `error` is being casted from its primitive `Error` to a more specific `error`.
+Having to write `catch let error as FooError` seems a bit inconsistent with the rest of how `catch` works, as `error` is inferred to be a constant of type `Error`. The following example makes it pretty clear.
+
+```swift
+do { ... }
+// Here we have to explicitly declare `error` as a constant.
+catch let error as FooError { ... }
+// Whereas here you have `error` for free.
+catch { ... }
+```
+
+This unbalanced behaviour comming from the same reserved word can induce confusion when writing down different specific and general catch clauses, having to declare error by themselves in one case and not in the other.

```

Comment:
This inconsistency can induce confusion when writing down different specific and general catch clauses, having to declare error by themselves in one case and omitting it in the other.